src/integrationTests/async.spec.ts

```diff
@@ -175,3 +195,165 @@ describe('when window.__uid2.init is called on SdkLoaded from a callback', () =>
+      const p = uid2.getFreshAdvertisingToken().catch((e: any) => {
+        expect(callback).toHaveBeenCalled();
+        throw e;
+      });
+      uid2.init({ callback: callback, identity: originalIdentity });
+      xhrMock.responseText = JSON.stringify({ status: 'error' });
+      xhrMock.onreadystatechange(new Event(''));
+      return expect(p).rejects.toBeInstanceOf(Error);
+    });
+  });
+
+  describe('when giving multiple promises', () => {
+    const identity = makeIdentity();
+    test('it should resolve all promises', () => {
+      const p1 = uid2.getFreshAdvertisingToken();
+      const p2 = uid2.getFreshAdvertisingToken();
+      const p3 = uid2.getFreshAdvertisingToken();
+      uid2.init({ callback: callback, identity: identity });
+      return expect(Promise.all([p1, p2, p3])).resolves.toStrictEqual(Array(3).fill(identity.advertising_token));
+    });
+  });
+})
+
+describe('when getFreshAdvertisingToken is called after init completed', () => {
+  describe('when initialised with a valid identity', () => {
+    const identity = makeIdentity();
+    test('it should resolve promise', () => {
+      uid2.init({ callback: callback, identity: identity });
+      return expect(uid2.getFreshAdvertisingToken()).resolves.toBe(identity.advertising_token);
+    });
+  });
+
+  describe('when a refreshing request inflight', () => {
+    const originalIdentity = makeIdentity({
+      refresh_from: Date.now() - 100000,
+    });
+    const updatedIdentity = makeIdentity({
+      advertising_token: 'updated_advertising_token'
+    });
+    test('it should resolve promise with updated advertising token', () => {
+      uid2.init({ callback: callback, identity: originalIdentity });
+      const p = uid2.getFreshAdvertisingToken().then((token: any) => {
+        expect(callback).toHaveBeenCalled();
+        return token;
+      });
+      xhrMock.responseText = btoa(JSON.stringify({ status: 'success', body: updatedIdentity }));
+      xhrMock.onreadystatechange(new Event(''));
+      return expect(p).resolves.toBe(updatedIdentity.advertising_token);
+    })
+  })
+
+  describe('when initialisation failed', () => {
+    test('it should reject promise', () => {
+      uid2.init({ callback: callback });
+      return expect(uid2.getFreshAdvertisingToken()).rejects.toBeInstanceOf(Error);
+    });
+  });
+
+  describe('when identity is temporarily not available', () => {
+    test('it should reject promise', () => {
+      const originalIdentity = makeIdentity({
+        refresh_from: Date.now() - 100000,
+        identity_expires: Date.now() - 1
+      });
+      uid2.init({ callback: callback, identity: originalIdentity });
+      xhrMock.responseText = JSON.stringify({ status: 'error' });
+      xhrMock.onreadystatechange(new Event(''));
+      return expect(uid2.getFreshAdvertisingToken()).rejects.toBeInstanceOf(Error);
+    });
+  });
+
+  describe('when disconnect() has been called', () => {
+    test('it should reject promise', () => {
```

Comment:
A lot of these tests are really just testing the same thing as the `getAdvertisingTokenAsync` tests - that's a lot of extra maintenance and code.